### PR TITLE
pass login_hint to OIDC connector's /auth endpoint in auth_code flow

### PR DESCRIPTION
**Pull request by acohen4**
#### Overview

To have dex pass the 'login_hint' query param to the OIDC connector, a few changes are made:
- a map of additional forwarded query parameters is passed as an  argument to the LoginURL connector interface. Each connector can choose whether to use these values.
- the OIDC connector decides on which forwarded query parameters to use by consulting a new field in the connector config called `forwardedLoginParams`
-  a login_hint column is added to the auth_request table to fully capture the request details

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Review by msteffen — APPROVED** [review]
If any of these (either this PR, or the previous PR breaking the `callback URI == issuer` hard-coded requirement) can be up-streamed, I think that could be good.

Actually, did you try making the other change into an upstream PR by any chance? If so, did anybody from Dex take a look/respond?

**Comment by acohen4** [comment]
> Actually, did you try making the other change into an upstream PR by any chance? If so, did anybody from Dex take a look/respond? Let me know if I can help

Great point. I haven't yet but will do so shortly with these two changes. I totally agree that it's valuable to try to get these changes upstreamed

**Comment by acohen4** [comment]
@msteffen after a deeper look at solving the issue for hub, they would also benefit from passing another query param that is specific to Auth0 but is not part of the OIDC standard. I have made another change here to pass along all the params for the connectors to consider using. 

I am conflicted as to whether the change data storage should be any different. I added the login_hint param to the authorization_request table since it's part of the OIDC spec, but don't think we should add other arbitrary fields outside the spec. This is definitely a point of feedback I'll run by the Dex team, but do you think I should be adding the login_hint to the auth_request table here?

File: connector/authproxy/authproxy.go
Comment (msteffen) [inline connector/authproxy/authproxy.go:40]: nit: would it make sense to use `map[string]string` instead of `url.Values` here, for consistency? (or, conversely, would it make sense to make the other signatures that ignore this argument accept `map[string]string` as well?)

File: connector/oidc/oidc.go
Comment (msteffen) [inline connector/oidc/oidc.go:200]: Nit: this is extremely superficial, but it might help get this PR approved if we left the space in (i.e. we want to match the surrounding code as much as possible—the Dex maintainers would be supporting whatever code we add, so ideally they should feel like they wrote it)

**Review by msteffen — APPROVED** [review]
Two minor nits, otherwise LGTM

File: connector/oidc/oidc.go
Comment (acohen4) [reply connector/oidc/oidc.go:200]: good point!

**Pull request merged**
